gradle/defaults-java.gradle

```diff
@@ -1,11 +1,51 @@
 // Configure Java project defaults.
 
-allprojects {
-  plugins.withType(JavaPlugin) {
+buildscript {
```

Comment:
I'd prefer if you separated the configuration and application of this plugin into a separate file (validation/errorprone.gradle)? Then each file configures one thing. Sure - there is an overhead in multiple passes over project collection but I think it's worth knowing what each particular file does and it makes them shorter.